### PR TITLE
Fix deprecation warnings on julia 0.4

### PR DESCRIPTION
**Pull request by maximsch2**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,4 @@
 julia 0.3
 Lazy
 MacroTools
+Compat
```
Comment (MikeInnes) [inline REQUIRE:4]: Maybe we can get rid of Compat and just bump Julia to 0.4?

Comment (maximsch2) [inline REQUIRE:4]: If we had `@compat` usage inside, that would certainly make sense (to make the code cleaner). As it is now, Compat just adds a couple of aliases (AbstractString=String) that make Hiccup work on 0.3 as well without any work from our side. So it seems logical to just leave it for now.

**Comment by maximsch2** [comment]
As a side note, tests don't work for me on 0.4 as well, this commit fixes them: https://github.com/maximsch2/Hiccup.jl/commit/dd9aa5a1b10cb904a57bf9b8596fdebe2e0d62af (was going to submit it after this got merged as they got entangled in my git repo. Cherry-picking it will also work)

File: REQUIRE (hunk 1 above)
Comment (MikeInnes) [inline REQUIRE:4]: Fair, I can get on board with that.

**Comment by MikeInnes** [comment]
Thanks for the patch – would definitely be interested in having the test fixes if you have a moment to throw up a PR.

**Pull request merged**